dependency_cache.go

```diff
@@ -88,6 +88,7 @@ func NewDependencyCache(context libcnb.BuildContext) (DependencyCache, error) {
 		UserAgent:         fmt.Sprintf("%s/%s", context.Buildpack.Info.ID, context.Buildpack.Info.Version),
 		Mappings:          map[string]string{},
 		DependencyMirrors: map[string]string{},
+		Logger:            bard.NewLogger(os.Stdout),
```

Comment:
Did you add this cause you were having issues with the tests?

The rationale is lost to time, but the Logger is usually set later by the caller after the dependency cache is created. See as [an example](https://github.com/paketo-buildpacks/spring-boot/blob/8501b4fe80f8f6f5a35c36de9a9c3c2b2f06c4d7/boot/build.go#L80-L84). We, unfortunately, can't change this cause it would break the API, but we will fix this in v2.

I ask cause I'd like to keep following that pattern (even though it's not great, consistency helps). If we set the logger here, what you've got will probably be OK (or in most cases overridden because of the pattern I explained), but I'd rather not push it. Then in v2, we'll clean all this up.

Reply:
The only reason I added this initialization was to print the warnings you suggested earlier in lines [168](https://github.com/bitgully/libpak/blob/1052fea6ed10479549808574f5f4ec32695de92d/dependency_cache.go#L168) and [181](https://github.com/bitgully/libpak/blob/1052fea6ed10479549808574f5f4ec32695de92d/dependency_cache.go#L181). Since these checks are happening during NewDependencyCache() the usual pattern of setting a logger afterwards is not working. Should we create a bard logger in setDependencyMirrors() instead? I'm happy for any suggestions here.

Reply:
Oh, ok. In that case, we don't have a choice. Let's just leave it this way. I don't think it will impact anything, I'm just going to add a comment so we don't forget why it's this way. Thanks!